src/output-json.c

```diff
@@ -739,6 +739,28 @@ json_t *CreateJSONHeader(const Packet *p, enum OutputJsonLogDirection dir,
         }
     }
 
+    /* mpls */
+    if (p->mpls_idx > 0) {
+        json_t *js_mpls;
+        switch (p->mpls_idx) {
```

Comment:
@jasonish this is the same as the vlan logic, but I'm not quite happy with that. I know we discussed this a while ago. Do you think we should go for an array in all cases here?

Reply:
Yes. I think if it can be an array, it should always be an array. I know JSON schema allows for both, but its just simpler if its always the same datatype.